### PR TITLE
Ensure consistent opacity for disabled form controls in all browsers

### PR DESCRIPTION
**Pull request by adamwathan**
We noticed that in Chrome, disabled `<select>` elements were being rendered at 70% opacity, but were being rendered at 100% opacity in Firefox and Safari. After some research, we learned that all form controls in all browsers are rendered at a lower opacity when disabled _if you don't include our other Preflight styles_, but some of the other normalizations we do in Preflight (specifically stuff around borders) causes each browser to abandon their default/native control styles, and only Chrome persists the user-agent opacity setting it was using for the untouched native control.

This PR just sets the opacity for all disabled elements to `1` so that there are no surprises and you can style your disabled elements however you want. This is something we ran into on Catalyst, and had to add `disabled:opacity-100` to our select controls to avoid since we had our own custom styles for them.

### CODE REVIEW

File: packages/@tailwindcss-postcss/src/__snapshots__/index.test.ts.snap [hunk 1]
```diff
@@ -475,6 +475,14 @@ exports[`\`@import 'tailwindcss'\` is replaced with the generated CSS 1`] = `
     color: inherit;
   }
 
+  ::file-selector-button:where() {
```
Comment (thecrypticace) [inline packages/@tailwindcss-postcss/src/__snapshots__/index.test.ts.snap:478]: This is definitely wrong. Maybe this is a lightningcss bug? Looks like it works in the playground so we should probably work on upgrading.

Maybe split the selector in preflight temporarily to work around it?

**Review by thecrypticace — CHANGES_REQUESTED** [review]
Noticed one thing that looks broken

File: packages/@tailwindcss-postcss/src/__snapshots__/index.test.ts.snap (hunk 1 above)
Comment (adamwathan) [reply packages/@tailwindcss-postcss/src/__snapshots__/index.test.ts.snap:478]: Ah yeah good catch. Noticing other stuff that's kinda in this same category of fixes, like `input` elements get a different background color when disabled too. Not sure how far to go with this stuff, tricky.

**Comment by philipp-spiess** [comment]
The changes in this PR are included in https://github.com/tailwindlabs/tailwindcss/pull/15036

**Pull request closed without merging**